server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/rfc/model/CreatedIds.scala

```diff
@@ -0,0 +1,55 @@
+/** **************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ * *
+ * http://www.apache.org/licenses/LICENSE-2.0                 *
+ * *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ * ***************************************************************/
+package org.apache.james.jmap.rfc.model
+
+import org.apache.james.jmap.rfc.model.CreatedIds.{ClientId, ServerId}
+import play.api.libs.json._
+
+case class CreatedIds(value: Map[ClientId, ServerId])
+
+object CreatedIds {
+  case class Id(value: String) {
+    require(Option(value).isDefined, "value cannot be null")
```

Comment:
IHMO your construct should be pure function and not throw exception.
This validation should be done in a factory method in its companion object if constructed from your application code.
For example : 
```
object Id {
  def of(value:String) : Try[Id] = ???
}
```
and when using it to construct a ClientId : ``` Id.of("myString").map(ClientId.apply)```


if it is constructing from Json deserialisation the validation should be handled in the `Read`

Reply:
the original Id class is implemented in my PR #3202. I'll fix it. BTW can't I use `apply()` instead of `of()` ?

Reply:
Seems like people are encouraging to use this pattern https://stackoverflow.com/a/1731144/6072746

But why returning `Try`? I find it more verbose, I had moved my check out of the constructor anyway... can you explain why still need `Try`?

Reply:
> the original Id class is implemented in my PR #3202. I'll fix it. BTW can't I use `apply()` instead of `of()` ?

If you do a apply with the same parameter of the companion object of the case class but a different return type. The compiler will not be happy.

Reply:
Honestly this check is not usefull if you interact with scala library like play-json.
if you get it from a java method which can return null. you will have to do : 
`Option(javaValue).map(Id.apply)` to get your Option of Id